src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ClassDataContract.cs

```diff
@@ -439,47 +439,50 @@ static internal bool IsNonAttributedTypeValidForSerialization(Type type)
             }
         }
 
-        private static string[] s_knownSerializableTypeNames = new string[] {
-                "System.Collections.Queue",
-                "System.Collections.Stack",
-                "System.Globalization.CultureInfo",
-                "System.Version",
-                "System.Collections.Generic.KeyValuePair`2",
-                "System.Collections.Generic.Queue`1",
-                "System.Collections.Generic.Stack`1",
-                "System.Collections.ObjectModel.ReadOnlyCollection`1",
-                "System.Collections.ObjectModel.ReadOnlyDictionary`2",
-                "System.Tuple`1",
-                "System.Tuple`2",
-                "System.Tuple`3",
-                "System.Tuple`4",
-                "System.Tuple`5",
-                "System.Tuple`6",
-                "System.Tuple`7",
-                "System.Tuple`8",
+        private static readonly Dictionary<string, string[]> s_knownSerializableTypeInfos = new Dictionary<string, string[]> {
+            { "System.Collections.Generic.KeyValuePair`2", Array.Empty<string>() },
+            { "System.Collections.Generic.Queue`1", new[] { "_syncRoot" } },
+            { "System.Collections.Generic.Stack`1", new[] {"_syncRoot" } },
+            { "System.Collections.ObjectModel.ReadOnlyCollection`1", new [] {"_syncRoot" } },
+            { "System.Collections.ObjectModel.ReadOnlyDictionary`2", new [] {"_syncRoot", "_keys","_values" } },
```

Comment:
Nit: for consistency, in the above two lines, `new []` can be `new[]` (without the space between `new` and `[]`).